src/centralserver/admin-service/src/main/resources/openapi-definition.yaml

```diff
@@ -1,357 +1,4105 @@
+          description: the url of the management services
+          example: https://dev.xroad.rocks/managementservice/manage/
+          format: url
+          minLength: 1
+          readOnly: true
+          type: string
+        wsdl_address:
+          description: the url of the management service wsdl
+          example: https://dev.xroad.rocks/managementservices.wsdl
+          format: url
+          minLength: 1
+          readOnly: true
+          type: string
+      required:
+        - security_server_id
+        - security_server_owners_global_group_code
+        - service_provider_name
+        - services_address
+        - wsdl_address
+        - service_provider_id
+      type: object
+    Member:
+      allOf:
+        - $ref: '#/components/schemas/MemberName'
+        - $ref: '#/components/schemas/MemberClassAndCode'
+      description: x-road member
+      properties:
+        id:
+          description: the encoded string representation of the x-road member id
+          example: FI:GOV:123:ABC
+          format: text
+          maxLength: 1023
+          minLength: 1
+          readOnly: true
+          type: string
+        instance_id:
+          description: Xroad instance id. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: FI
+          format: text
+          maxLength: 255
+          minLength: 1
+          readOnly: true
+          type: string
+        member_type:
+          $ref: '#/components/schemas/MemberType'
+        subsystem_code:
+          description: Subsystem code. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: ABC
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      type: object
+    MemberClass:
+      allOf:
+        - $ref: '#/components/schemas/MemberClassDescription'
+      description: Central server's member classes
+      properties:
+        code:
+          description: member class code
+          example: ORG
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      required:
+        - code
+      type: object
+    MemberClassAndCode:
+      description: x-road member class and code
+      properties:
+        member_class:
+          description: Member class. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: GOV
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+        member_code:
+          description: Member code. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: 123
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      required:
+        - member_class
+        - member_code
+      type: object
+    MemberClassDescription:
```

Comment:
At first glance this seems quite fine grained modeling. Would have been tempted to just have this as a simple direct property of `/MemberClass`, and work with `required`/`readOnly`, etc definitions, and have `updateMemberClassDescription` take in a simple string.

It does not seem to create any immediately visible big issues, but can you still explain the thinking behind this instead of simple string for the property, since it is so widely used pattern?

Reply:
It removes duplication. The `MemberClassDescription` can be used as a part of the `MemberClass` model instead of declaring a new string with the same description and constraints. 

My thought process was that if you can patch update a resource (in this case the `MemberClass.description` property) via the API, then separate it into its own model so that it can be used as a request body later.

This method also pushes into creating API models that have properties so that they can directly be used as a request body (the class code can never be modified)
```
PATCH /member-classes/SomeNiceClassCode

{
  "description": "New description!"
}
```

Reply:
All right, seems sensible :+1:

Reply:
From my recent work, experiences:

- constructing schema by combining objects that are only a single field can seem a bit too fine grained. If there's an update method for that single field, probably makes sense, if there's not, it might not be a good choice
- sometimes property names differ: `description` may need to be `member-class-description` in some other combo-object
- tooling support for allOf can vary; at least postman did not understand all constructs that we had used (although the issue was not critical)